source/module_cell/klist.h

```diff
@@ -29,60 +102,121 @@ class K_Vectors
+    static std::vector<int> kspacing_tompmesh(const ModuleBase::Matrix3& bmat,                  //< [in] reciprocal lattice matrix
+                                       const double& lat0,                               //< [in] lattice_constant
+                                       const std::vector<double>& kspacing);             //< [in] kspacing
+    // write Monkhorst-Pack kpoints to KPT file
+    static std::string write_abacus_mpkmesh(const std::string& center,         //< [in] can be "Gamma" or "Monkhorst-Pack"
+                                     const std::vector<int>& nmp,       //< [in] number of Monkhorst-Pack kpoints
+                                     const std::vector<int>& shifts);   //< [in] shifts of Monkhorst-Pack kpoints
+    // write Line mode kpoints to KPT file
+    static std::string write_abacus_kline(const std::string& scale,                             //< [in] can be "Direct" or "Cartesian"
+                                          const std::vector<std::vector<double>>& kvec,         //< [in] kpoints coordinates
+                                          const std::vector<int>& nks);                         //< [in] number of kpoints in each segment
+    // overloaded version for ModuleBase::Vector3, which is not safe and has partially been deprecated by new compilers like icpx
+    static std::string write_abacus_kline(const std::string& scale,                             //< [in] can be "Direct" or "Cartesian"
+                                          const std::vector<ModuleBase::Vector3<double>>& kvec, //< [in] kpoints coordinates
+                                          const std::vector<int>& nks);                         //< [in] number of kpoints in each segment
 
-    int get_nkstot_full() const
-    {
-        return this->nkstot_full;
-    }
+    // Synchronize
+    // synchronize between alpha and beta spin. Because kpoint does not distinguish spin physically, this way
+    // is just a computational implementation convention, rather than physically strict/correct way.
+    void sync_kvec_betweenspin(const int& nspin); // because in principle there is forever 1 set of kpoints,
+                                                  // the special treatment like treating different spin
+                                                  // kpoints separately is not physically correct. Thus the
+                                                  // nspin is set as one parameter instead of a member variable.
+    // synchronize between kvec_c and kvec_d
+    void sync_kvec_betweencd(const bool& direct,            //< [in] true is from kvec_d to kvec_c, false is from kvec_c to kvec_d
+                             const ModuleBase::Matrix3& t); //< [in] transformation matrix
+    // synchronize between MPI processes
+    void sync_kvec_betweenproc();
 
-    void set_nks(int value) {
-        this->nks = value;
+    std::vector<std::vector<double>> kvec(const std::vector<int>& iks,
```

Comment:
what are these two functions (kvec(...)) used for? They are not been called anywhere.